### PR TITLE
Add kserve-modelmesh-serving.

### PR DESCRIPTION
**Pull request by wlynch**
Related: https://github.com/chainguard-dev/image-requests/issues/3818

#### For new package PRs only

- [x] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

### CODE REVIEW

File: kserve-modelmesh-serving.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package:
+  name: kserve-modelmesh-serving
+  version: 0.12.0
+  epoch: 0
+  description: ModelMesh Serving is the Controller for managing ModelMesh, a general-purpose model serving management/routing layer.
+  copyright:
+    - license: Apache-2.0
+
+environment:
```
Comment (developer-guy) [inline kserve-modelmesh-serving.yaml:9]: you don't need any of these o7

File: kserve-modelmesh-serving.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      repository: https://github.com/kserve/modelmesh-serving.git
+      tag: v${{package.version}}
+      expected-commit: d0e56fa4ac7547e644c87d63dc97a14f612c0391
+
+  - uses: go/build
+    with:
+      packages: .
+      output: manager
+
+  - uses: strip
```
Comment (developer-guy) [inline kserve-modelmesh-serving.yaml:28]: I think this one also not needed, go/build uses a "-w" flag to strip the binary, afaik.

**Review by developer-guy — DISMISSED** [review]

File: kserve-modelmesh-serving.yaml (hunk 1 above)
Comment (wlynch) [reply kserve-modelmesh-serving.yaml]: Done.

**Comment by jamonation** [comment]
Could this benefit from using the `kwok` test pipeline? The [helm-operator package](https://github.com/wolfi-dev/os/blob/2e718418f25c851dd05271ea014bd3342797b51a/helm-operator.yaml#L49-L67) uses it to good effect. 

At a minimum it would be helpful to add a package level test to validate the binary runs, and ideally a more complete functional test like the kwok example. That said, for this there might need to be some configMap loading, so if it gets too complicated stubbing things out it might not be worth it.

**Comment by wlynch** [comment]
@jamonation Can you point me to more documentation on that? I don't see any mention of it in the melange docs or the wiki.

**Comment by jamonation** [comment]
There's no doc on how to use it @wlynch - but there are a few packages here in Wolfi to crib from as examples. @joshrwolf added the pipeline I believe, so it might be worth asking him for more guidance if there aren't any good examples that would work with this package.

**Review by joshrwolf — APPROVED** [review]
I LOVE THESE TESTS

**Pull request merged**